### PR TITLE
Multiple endpoints

### PR DESCRIPTION
**Pull request by Alan-Cha**

### CODE REVIEW

File: charts/iter8/templates/_task-grpc.tpl [hunk 1]
```diff
@@ -3,41 +3,68 @@
...
 {{- if $vals.metadataURL }}
 # task: download metadata JSON file from URL
 - run: |
     curl -o metadata.json {{ $vals.metadataURL }}
-{{- $pf := dict "metadata-file" "metadata.json" }}
-{{- $vals = mustMerge $pf $vals }}
+{{- $_ := set $vals "metadata-file" "metadata.json" }}
+{{- end }}
+# handle endpoints
+{{- range $endpointID, $endpoint := $vals.endpoints }}
```
Comment (kalantar) [inline charts/iter8/templates/_task-grpc.tpl:39]: I think we need a check that `endpoint` contains `host` and `call` (or that it is in `$_`). The earlier check only tested that `endpoints` existed but not the `host` and `call` are defined for each.

**Review by kalantar — CHANGES_REQUESTED** [review]

File: charts/iter8/templates/_task-grpc.tpl (hunk 1 above)
Comment (sriumcp) [reply charts/iter8/templates/_task-grpc.tpl:39]: We can keep what the check that's currently in the PR. In addition, we could introduce an additional check for each endpoint (i.e., either this endpoint explicitly specifies host/call, or it is part of the common parameter).

File: charts/iter8/templates/_task-grpc.tpl [hunk 2]
```diff
@@ -3,41 +3,85 @@
...
-{{- $vals = mustMerge $pf $vals }}
+{{- $_ := set $vals "data-file" "data.json" }}
 {{- end }}
 {{- if $vals.binaryDataURL }}
 # task: download binary data file from URL
 - run: |
     curl -o data.bin {{ $vals.binaryDataURL }}
-{{- $pf := dict "binary-file" "data.bin" }}
-{{- $vals = mustMerge $pf $vals }}
+{{- $_ := set $vals "binary-file" "data.bin" }}
```
Comment (Alan-Cha) [inline charts/iter8/templates/_task-grpc.tpl:48]: New pattern?

**Review by kalantar — CHANGES_REQUESTED** [review]
The warmup PR (already merged) also changed these files. Is there any impact? Also, the version (in `Chart.yaml` should be updated.

**Review by sriumcp — CHANGES_REQUESTED** [review]
Update chart version

**Review by sriumcp — APPROVED** [review]
lgtm

**Review by kalantar — APPROVED** [review]

**Pull request merged**